### ISSUE TITLE
Domains - Broken path to Domain Settings for domain-only hosted outside of WordPress.com

### ISSUE THREAD

**Issue by formosattic** [Automattic/wp-calypso#88451]
### Quick summary

For domain-only registered at WordPress.com but hosted elsewhere the path `All Domains > Domain > View Settings` is broken.

User report: 7816442-zd-a8c

### Steps to reproduce

I haven't tested it for myself, but here's how to reproduce the user's config:

1. Register a domain-only at WordPress.com
2. Edit the DNS so the domain points to another host (e.g.: Pressable)
3. Go to the `All Domains` page, https://wordpress.com/domains/manage
4. Look for the domain, then click the three dots and select `View settings`

### What happened

The link redirected to a page with this message:

> Domains are not available for this site.
> You can only purchase domains for sites hosted on WordPress.com at this time.

Dead link pattern:
`https://wordpress.com/domains/manage/all/examplesite.com/edit/examplesite.com`

![image](https://github.com/Automattic/bugomattic/assets/15944096/6228c04f-b181-4bb7-958e-62a60149353f)

### What's expected & Workaround

It should redirected to a page to edit the domain settings.

I'm not sure how it works with domains only hosted outside of WordPress.com, but I found a working link by going to:

1. `Purchases`, https://wordpress.com/me/purchases
2. Look for the Domain Registration upgrade of the domain, then click on it
3. Once in `Purchase Settings`, click on the `Domain Settings` link

Good working link pattern:
`https://wordpress.com/domains/manage/examplesite.com/edit/examplesite.wordpress.com`

### Other notes

Zooming out a bit, in addition of fixing the link on the `All Domains` page, we could also have a better integrated path by adding the Domains option under the Upgrades (only for domain-only registered at WordPress.com), similar to how it is for sites hosted at WordPress.com.

| Site hosted at WordPress.com | Site hosted elsewhere |
| ------------- | ------------- |
| ![image-1](https://github.com/Automattic/bugomattic/assets/15944096/9bee8c49-5eed-4609-9918-a8d7a9915cee) | Can we add a path to `Upgrades > Domains`? ![image-1](https://github.com/Automattic/bugomattic/assets/15944096/a0a2c226-646d-4fff-8bf9-20d5fc81d8e5)  |

**Comment by rickmgithub** [comment]
## 📌 REPRODUCTION RESULTS ##
Not able to replicate.
Purchased a domain _ewfewfewfewfewefwo93o7yr329.blog_ and followed the steps but got to the settings page.


## 📌 ACTIONS ##
- Author feedback

##Message to author##
Thanks so much for the report. I couldn't replicate the issue with the same set up on a domain purchased at WordPress.com. When you mentioned "I'm not sure how it works with domains only hosted outside of WordPress.com," can you tell us more about this so we can test it? Do you mean one that has been transferred in?

For now I've closed the report but if you have additional information please do reopen it.

**Comment by formosattic** [comment]
@rickmgithub , thanks for testing. I see your test domain was pointing to the other host via nameservers.
I checked the ticket again, 7816442-zd-a8c , the domain was pointing to another host with A records, I didn't think it'd matter, but maybe it does?

**Comment by liviopv** [comment]
@formosattic I triaged this again and I wasn't able to reproduce it either, even when pointing the domain to external A Records

What I did:

1. Turned liviopv.com into a domain-only site
2. Switched the A Record to 34.211.180.66
3. I could still see the domain-only site in /domains/manage
![CleanShot 2024-03-13 at 17 14 28@2x](https://github.com/Automattic/wp-calypso/assets/14153300/03279591-f719-4a5c-9bd3-0362543e1bac)

I tested the above unproxied and with a non-a11n account.

This particular flow - domain-only sites pointing to external IPs - has been a common one since the Google Domains takeover project, so if it was broken or a regression had been introduced, we would have had other reports by now considering how many similar domains exist with that same setup.

Closing this for now.
